zettarepl/replication/run.py

```diff
@@ -264,19 +286,51 @@ def run_replication_steps(step_templates: [ReplicationStepTemplate], observer=No
 
         incremental_base, snapshots = get_snapshots_to_send(src_snapshots, dst_snapshots,
                                                             step_template.replication_task)
-        if incremental_base is None and dst_snapshots:
-            if step_template.replication_task.allow_from_scratch:
-                logger.warning("No incremental base for replication task %r on dataset %r, destroying all destination "
-                               "snapshots", step_template.replication_task.id, step_template.src_dataset)
-                destroy_snapshots(
-                    step_template.dst_context.shell,
-                    [Snapshot(step_template.dst_dataset, name) for name in dst_snapshots]
-                )
+        if incremental_base is None:
+            if dst_snapshots:
+                if step_template.replication_task.allow_from_scratch:
+                    logger.warning(
+                        "No incremental base for replication task %r on dataset %r, destroying all destination "
+                        "snapshots", step_template.replication_task.id, step_template.src_dataset,
+                    )
+                    destroy_snapshots(
+                        step_template.dst_context.shell,
+                        [Snapshot(step_template.dst_dataset, name) for name in dst_snapshots]
+                    )
+                else:
+                    raise NoIncrementalBaseReplicationError(
+                        f"No incremental base on dataset {step_template.src_dataset!r} and replication from scratch "
+                        f"is not allowed"
+                    )
             else:
-                raise NoIncrementalBaseReplicationError(
-                    f"No incremental base on dataset {step_template.src_dataset!r} and replication from scratch "
-                    f"is not allowed"
-                )
+                if not step_template.replication_task.allow_from_scratch:
+                    if is_immediate_target_dataset:
+                        # We are only interested in checking target datasets, not their children
+                        try:
+                            dst_type = get_property(step_template.dst_context.shell, step_template.dst_dataset, "type")
```

Comment:
Doesnt that make a remote call?

is there any benefit in getting all three at once so we dont make two separate calls? zfs get type,used,refer ?